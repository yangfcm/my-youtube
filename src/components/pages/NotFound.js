import React from "react";
import { Link } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import Error from "../common/ErrorMessage";

class NotFound extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Error message="The page you requested doesn't exist" />
        <div className="text-center">
          <Link to="/" className="btn btn-outline-info">
            <FontAwesomeIcon icon={"home"} />
            {"   "}
            Back to Homepage
          </Link>
        </div>
      </React.Fragment>
    );
  }
}

export default NotFound;
