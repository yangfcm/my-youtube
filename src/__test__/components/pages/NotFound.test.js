import React from "react";
import { shallow } from "enzyme";
import { Link } from "react-router-dom";
import NotFound from "components/pages/NotFound";
import Error from "components/common/ErrorMessage";

describe("Test NotFound page", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<NotFound />);
  });
  afterEach(() => {
    wrapper.unmount();
  });

  it("should render component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render Error component", () => {
    expect(wrapper.find(Error).exists()).toBe(true);
    expect(wrapper.find(Error).prop("error")).toEqual({
      displayMessage: "The page you requested doesn't exist",
    });
  });

  it("should render Link and it navigates to homepage", () => {
    const link = wrapper.find(Link);
    expect(link.exists()).toBe(true);
    expect(link.prop("to")).toBe("/");
    expect(link.text()).toContain("Back to Homepage");
  });
});
