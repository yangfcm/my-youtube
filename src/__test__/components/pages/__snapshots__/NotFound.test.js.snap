// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NotFound page should render component 1`] = `
<Fragment>
  <ErrorMessage
    error={
      Object {
        "displayMessage": "The page you requested doesn't exist",
      }
    }
  />
  <div
    className="text-center"
  >
    <Link
      className="btn btn-outline-info"
      to="/"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="home"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
         
      Back to Homepage
    </Link>
  </div>
</Fragment>
`;
